packages/graz/src/hooks/methods.ts

```diff
@@ -99,3 +115,191 @@ export const useSendIbcTokens = ({
     status: mutation.status,
   };
 };
+
+export type UseInstantiateContractArgs<Message extends Record<string, unknown>> = {
+  codeId: number;
+} & MutationEventArgs<InstantiateContractMutationArgs<Message>, InstantiateResult>;
+
+/**
+ * graz mutation hook to instantiate a CosmWasm smart contract when supported.
+ *
+ * @example
+ * ```ts
+ * import { useInstantiateContract } from "graz"
+ *
+ * const { instantiateContract: instantiateMyContract } = useInstantiateContract({
+ *   codeId: 4,
+ *   onSuccess: ({ contractAddress }) => console.log('Address:', contractAddress)
+ * })
+ *
+ * const instantiateMessage = { foo: 'bar' };
+ * instantiateMyContract(instantiateMessage);
+ * ```
+ */
+export const useInstantiateContract = <Message extends Record<string, unknown>>({
+  codeId,
+  onError,
+  onLoading,
+  onSuccess,
+}: UseInstantiateContractArgs<Message>) => {
+  const { data: account } = useAccount();
+  const accountAddress = account?.bech32Address;
+
+  const mutationFn = (args: InstantiateContractMutationArgs<Message>) => {
+    const contractArgs: InstantiateContractArgs<Message> = {
+      ...args,
+      fee: args.fee ?? "auto",
+      senderAddress: accountAddress,
+      codeId,
+    };
+
+    return instantiateContract(contractArgs);
+  };
+
+  const queryKey = ["USE_INSTANTIATE_CONTRACT", onError, onLoading, onSuccess, codeId];
```

Comment:
since `accountAddress` is a dependant on this function let's add it to the key
```suggestion
  const queryKey = ["USE_INSTANTIATE_CONTRACT", onError, onLoading, onSuccess, codeId, accountAddress];
```